src/main/java/org/sonarlint/intellij/config/project/ExclusionTable.java

```diff
@@ -0,0 +1,196 @@
+    typeColumn.setPreferredWidth(width);
+    typeColumn.setMaxWidth(width);
+    typeColumn.setMinWidth(width);
+
+    table.getTableHeader().setReorderingAllowed(false);
+    table.getSelectionModel().setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
+
+    table.addMouseListener(new MouseAdapter() {
+      @Override
+      public void mouseClicked(MouseEvent evt) {
+        if (evt.getClickCount() == 2) {
+          editEntry();
+        }
+      }
+    });
+
+    ToolbarDecorator toolbarDecorator = ToolbarDecorator.createDecorator(table)
+      .setEditActionName("Edit")
+      .setEditAction(e -> editEntry())
+      .setAddAction(new AddEntryAction())
+      .setRemoveAction(new RemoveEntryAction())
+      .disableUpDownActions();
+
+    panel = new JPanel(new BorderLayout());
+    panel.add(toolbarDecorator.createPanel(), BorderLayout.CENTER);
+  }
+
+  public JPanel getComponent() {
+    return panel;
+  }
+
+  public void set(List<ExclusionItem> data) {
+    model.set(data);
+  }
+
+  public List<ExclusionItem> get() {
+    return new ArrayList<>(model.items());
+  }
+
+  private void editEntry() {
+    int selectedIndex = table.getSelectedRow();
+    if (selectedIndex >= 0) {
+      ExclusionItem value = model.items().get(selectedIndex);
+      ExclusionItem newValue = onEdit.apply(value);
+      if (newValue != null) {
+        model.items().set(selectedIndex, newValue);
+      }
+    }
+  }
+
+  private class AddEntryAction implements AnActionButtonRunnable {
+    @Override
+    public void run(AnActionButton anActionButton) {
+      ExclusionItem input = onAdd.get();
+      if (input != null) {
+        model.items().add(input);
+        int lastRow = model.getRowCount() - 1;
+        table.setRowSelectionInterval(lastRow, lastRow);
+      }
+    }
+  }
+
+  private static class Model extends AbstractTableModel {
+    private List<ExclusionItem> rows = new ArrayList<>();
+
+    @Override public int getRowCount() {
+      return rows.size();
+    }
+
+    @Override public int getColumnCount() {
+      return 2;
+    }
+
+    public void add(ExclusionItem item) {
+      int lastRow = rows.size();
+      rows.add(item);
+      fireTableRowsInserted(lastRow, lastRow);
+    }
+
+    public void remove(int index) {
+      rows.remove(index);
+      fireTableRowsDeleted(index, index);
+    }
+
+    @Override
+    public boolean isCellEditable(int row, int column) {
+      return false;
+    }
+
+    @Override
+    public String getColumnName(int column) {
+      return (column == 0) ? "Type" : "Item";
+    }
+
+    public void set(List<ExclusionItem> rows) {
+      this.rows = rows;
+      fireTableDataChanged();
+    }
+
+    public List<ExclusionItem> items() {
+      return rows;
+    }
+
+    @Override public Object getValueAt(int rowIndex, int columnIndex) {
+      ExclusionItem item = rows.get(rowIndex);
+      if (columnIndex == 0) {
+        return StringUtils.capitalize(item.type().name().toLowerCase());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define the locale to be used in this String operation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1449)